tests/fsharp/Compiler/ConstraintSolver/PrimitiveConstraints.fs

```diff
@@ -0,0 +1,109 @@
+﻿// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+
+namespace FSharp.Compiler.UnitTests
+
+open NUnit.Framework
+open FSharp.Compiler.SourceCodeServices
+
+[<TestFixture>]
+module PrimitiveConstraints =
+
+    [<Test>]
+    let ``Test primitive : constraints``() =
+        CompilerAssert.CompileExeAndRun
+            """
+#light
+
+type Foo(x : int) =
+    member   this.Value      = x
+    override this.ToString() = "Foo"
+
+type Bar(x : int) =
+    inherit Foo(-1)
+    member   this.Value2     = x
+    override this.ToString() = "Bar"
+
+let test1 (x : Foo) = x.Value
+let test2 (x : Bar) = (x.Value, x.Value2)
+
+let f = new Foo(128)
+let b = new Bar(256)
+
+if test1 f <> 128       then exit 1
+if test2 b <> (-1, 256) then exit 1
+"""
+
+    [<Test>]
+    let ``Test primitive :> constraints``() =
+        CompilerAssert.CompileExeAndRun
+            """
+#light
+type Foo(x : int) =
+    member   this.Value      = x
+    override this.ToString() = "Foo"
+
+type Bar(x : int) =
+    inherit Foo(-1)
+    member   this.Value2     = x
+    override this.ToString() = "Bar"
+
+type Ram(x : int) =
+    inherit Foo(10)
+    member   this.ValueA     = x
+    override this.ToString() = "Ram"
+
+let test (x : Foo) = (x.Value, x.ToString())
+
+let f = new Foo(128)
+let b = new Bar(256)
+let r = new Ram(314)
+
+if test f <> (128, "Foo") then exit 1
+if test b <> (-1, "Bar") then exit 1
+if test r <> (10, "Ram") then exit 1
+"""
+
+    [<Test>]
+    let ``Test primitive : null constraint``() =
```

Comment:
I think we could modify this test. Since we're not testing that the code _runs_ as expected, we're really just interested in verifying that the `null` constraint applies. So we'd like to verify that `isNull x` will compile if `x` has `null` as a proper value, and that it does not compile of `x` does not have `null` as a proper value.

This will incidentally also remove the need for `exit`.